src/Mealz/MealBundle/Entity/Participant.php

```diff
@@ -187,9 +186,24 @@ public function isPending(): bool
         return 0 !== $this->getOfferedAt();
     }
 
-    public function getCombinedDishes(): Collection
+    public function getCombinedDishes(): DishCollection
     {
-        return $this->combinedDishes;
+        if (null === $this->combinedDishes) {
+            return new DishCollection();
+        }
+
+        return new DishCollection($this->combinedDishes->toArray());
+    }
+
+    public function setCombinedDishes(?DishCollection $collection): void
+    {
+        $this->combinedDishes->clear();
```

Comment:
Cleared and in 206 overriden, but why?

Reply:
What happens if you call setCombinedDishes(...) of a fresh participant from doctrine where $this->combinedDishes is null?

Reply:
"clear" is used to delete participant_dish relations. I had a problem where these relations didn't delete. I changed it a bit now.

Reply:
I see, but for this you need to check if $this->combinedDishes isn't null. Otherwise it's possible that you invoke clear on a null object (loaded initially by doctrine). To avoid this check you could set $this->combinedDishes to null.